server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -133,6 +132,21 @@ private void testPeerRecovery(int numberOfIterations, boolean invokeFlush) throw
         );
     }
 
+    public void testRemoteStoreIndexCreationAndDeletionWithReferencedStore() throws InterruptedException, ExecutionException {
```

Comment:
Change looks good and it makes sense to have a separate lock for remote store. Just wondering why assertion is only failing in delete index flow only. Can it fail if shard is moved away from this node as well?

Reply:
It fails only in delete index flow as we close `remoteStore` only if index is deleted.

https://github.com/opensearch-project/OpenSearch/blob/58d1164f74e921a26b7a73b6185b38cc87bbc7a9/server/src/main/java/org/opensearch/index/IndexService.java#L795-L797